tuf/repository_lib.py

```diff
@@ -1458,9 +1489,26 @@ def generate_snapshot_metadata(metadata_directory, version, expiration_date,
       # list these roles found in the metadata directory.
       if tuf.roledb.role_exists(rolename, repository_name) and \
           rolename not in ['root', 'snapshot', 'timestamp', 'targets']:
-        fileinfodict[metadata_name] = get_metadata_versioninfo(rolename,
+
+        length = None
+        hashes = None
+        # We want to make sure we are calculating length and hashes only when
+        # at least one of them is needed. Otherwise, for adoptors of tuf with
+        # lots of delegations, this will cause unnecessary overhead.
+        if use_length or use_hashes:
```

Comment:
Thanks for adding this! Ideally we would only calculate length when `use_length` and hashes when `use_hashes`, but that will require some additions to securesystemslib and as such I think we should open an Issue to track that enhancement once this PR has landed, linked to the corresponding required API additions in securesystemslib https://github.com/secure-systems-lab/securesystemslib/issues/246